### PR TITLE
Term entry dart list.index of

### LINKED ISSUE #4307: [Term Entry] Dart List .indexOf()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List `.indexOf()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/indexOf/indexOf.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by AgnesaTb**
### Description

- Term Entry Dart list.indexOf()
- Closes #4307 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by Sriparno08 — COMMENTED** [review]
Hey @AgnesaTb, thanks for contributing to Codecademy!

I've listed all the modifications that you need to make in your entry below:

**Review by Sriparno08 — APPROVED** [review]
Looks good for a second review 👍

File: content/dart/concepts/list/terms/indexOf/indexOf.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+Title: '.indexOf()'
+Description: 'Returns the first occurrence of the indicated element in a list.'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/indexOf/indexOf.md:3]:
```suggestion
Description: 'Returns the first occurrence of the given element in a list.'
```

File: content/dart/concepts/list/terms/indexOf/indexOf.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ - 'Dart'
+ - 'Methods'
+ - 'Lists'
+ - 'Index'
+CatalogContent:
+ - 'learn-dart'
+ - 'paths/computer-science'
+---
+
+In Dart, the **`.indexOf()`** method returns the first occurrence of the indicated element in a list. If the element is not found, the method will return `-1`.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/indexOf/indexOf.md:17]:
```suggestion
In Dart, the **`.indexOf()`** method returns the first occurrence of the given element in a list. If the element is not found, the method will return `-1`.
```

**Review by avdhoottt — APPROVED** [review]
LGTM!!

**Pull request merged**